<h1>İçindekiler<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#İstatistik-nedir?" data-toc-modified-id="İstatistik-nedir?-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>İstatistik nedir?</a></span></li><li><span><a href="#Temel-istatistiksel-parametreler-ve-yöntemler" data-toc-modified-id="Temel-istatistiksel-parametreler-ve-yöntemler-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Temel istatistiksel parametreler ve yöntemler</a></span></li><li><span><a href="#Hipotez-testi-nedir?" data-toc-modified-id="Hipotez-testi-nedir?-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Hipotez testi nedir?</a></span></li></ul></div>

## İstatistik nedir?

Düşünün ki elimizde çok büyük bir veri seti var, mesela şehrinizdeki tüm insanların boy, kilo ve yaş bilgilerine sahipsiniz. Bir insanının tüm bu ham veriyi tümüyle anlayabilmesi mümkün değildir. Bu sebeple veriyi anlayabilmemizi kolaylaştıracak çeşitli metrikler ve yöntemler kullanmamız gerekmektedir. Örneğin bu verinin ortalamasına bakabiliriz. Böylece insanların ortalama ne kadar bir boya sahip oldukları konusunda fikrimiz olur. Veya en kısa ve en uzun boylu insanlara bakıp bunları ortalama ile karşılaştırabiliriz.  

**İstatistik** en genel tanımıyla elimizdeki veriyi anlayabilmek için kullandığımız matematiksel yöntemler ve nümerik teknikler bütünüdür. 

Bu kısa tanımın arkasında çok büyük bir araştırma ve bilimsel geçmiş bulunmaktadır. Bu yazımızda bu geniş alana ilişkin bilinmesi gereken çok temel bilgileri sizlere sunmaya çalışacağız. 

## Temel istatistiksel parametreler ve yöntemler

Hangi alanda çalışırsanız çalışın, temel bilgileri hakkıyla öğrenmeden ilerlemeniz mümkün değildir. İşe temel istatistiksel kavramları ve hesaplamaları gözden geçirerek başlayalım. Kavramları gözden geçirebilmek için örnek bir veri seti kullanalım isterseniz.

Aşağıdaki adresteki veri seti şu aşamada yeterli gibi duruyor. Bunu "csv" halinde indirip kullanabiliriz.

https://www.kaggle.com/yersever/500-person-gender-height-weight-bodymassindex

Not: Veriyi *pandas* kullanarak okuyacağız. Eğer bu konuda yardıma ihtiyacınız varsa *pandas* kütüphanesi kullanımı ile ilgili yazımıza bakabilirsiniz.

In [41]:
import pandas as pd # pandas dataframe yapısını kullanmak için gerekecek
import numpy as np # np lazım olabilir :)
sample_data = pd.read_csv("500_Person_Gender_Height_Weight_Index.csv")

In [42]:
# verimizdeki sütunlara bakalım
sample_data.columns

Index(['Gender', 'Height', 'Weight', 'Index'], dtype='object')

Her satırda bir kişiye ilişkin "Cinsiyet, Boy ve Kilo" bilgilerinin olduğunu görüyoruz. Daha rahat çalışabilmek için bunları  Boy ve Kilo değişkenlerine atayalım.

In [43]:
boy = sample_data[["Height"]].values.ravel()
kilo = sample_data[["Weight"]].values.ravel()
# sondaki values fonksiyonu değerleri almak için, ravel fonksiyonu 
# ise ortaya çıkan array'i bitişik yazmak için kullanılmıştır.

>**Ortalama (mean):** Ortalama bir verideki tüm elemanların toplanıp bunun eleman sayısına bölünmesiyle bulunur. 

Örnek sayısına $n$ dersek ortalama değerin matematiksel ifadesi aşağıdaki gibidir:

$${\displaystyle {\bar {x}}={\frac {1}{n}}\left(\sum _{i=1}^{n}{x_{i}}\right)}$$


Gelin hep beraber fonksiyon kullanmak yerine yukarıdaki denklemi bir fonksiyon cinsinden ifade edelim

In [44]:
def ortalama(x):
    # sum toplamı bulur
    # len eleman sayısını bulur
    return sum(x) / len(x)

In [45]:
# Yazıdığımız fonksiyonu kullanarak
# Boy (Height) değişkeninin ortalama değerini bulalım
print(ortalama(boy))

169.944


Bu örneklemdeki boy ortalamasının 169.944 cm olduğunu görüyoruz.

>**Medyan (median):** Bir dizideki değerleri sıraladığımız zaman en ortadaki değere medyan denilmektedir.Eğer dizideki eleman sayısı çift ise en ortadaki iki elemanın ortalaması alınarak hesaplanır.

In [46]:
def medyan(x):
    # toplam örnek sayısını bulalım
    n = len(x)
    # diziyi sıralayalım
    x = sorted(x)
    # tam sayı bölme yaparak orta noktayı buluyoruz
    orta_nokta = n // 2
    
    # eğer örnek sayısı tek ise en ortadaki değer döndürülüyor
    if n % 2 == 1:
        return x[midpoint]
    # eğer çift ise bir öncek ve sonraki değerin ortalaması alınır
    else:
        return (x[orta_nokta-1] + x[orta_nokta]) / 2

In [50]:
print(medyan(boy))

170.5


## Hipotez testi nedir?
Çoğu zaman elimizde bulunan veriyi kullanarak çeşitli çıkarımlar yapmaya çalışırız. Yapmak istediğimiz çıkarımlar genellikle istatistiksel parametreler üzerine olur. 

Örneğin bir şehirde yaşayan insanların **ortalama boylarını** bulmaya çalıştığımızı varsayalım. Fakat şehrimizde 1 milyon insan var ve bu insanların hepsinin boylarını bilmemiz çok olanaklı değil. Bu amaçla bir anket yapmaya karar veriyoruz ve  yaklaşık 1000 kişiye sorup boylarını kaydediyoruz. 

Burada kendimize şu temel soruyu sormamız gerekir: Elde ettiğimiz 1000 kişinin boy verisini kullanarak şehrin tümünün boy ortalamasını bulmamız mümkün müdür?